extensions/SharkPool/Spotify.js

```diff
@@ -0,0 +1,506 @@
+// Name: Spotify API
+// ID: SPspotify
+// Description: Fetch Statistics and Information and Play Songs from Spotify
+// By: SharkPool
+
+// Version 1.1.0
+
+(function (Scratch) {
+  "use strict";
+  if (!Scratch.extensions.unsandboxed) throw new Error("Spotify API must run unsandboxed");
+
+  const menuIconURI = "https://upload.wikimedia.org/wikipedia/commons/8/84/Spotify_icon.svg";
+  const blockIconURI = "https://uxwing.com/wp-content/themes/uxwing/download/brands-and-social-media/spotify-white-icon.svg";
+
+  const proxy = "https://api.codetabs.com/v1/proxy?quest=";
+  let audioInstances = [];
+  let lastFetchedSong = [];
+  let lastFetchedArtist = [];
+  let lastFetchedPlaylist = [];
+
+  class SPspotify {
+    constructor() {
+      this.audioContext = new AudioContext();
+      this.gainNode = this.audioContext.createGain();
+      this.gainNode.gain.value = 1;
+      this.gainNode.connect(this.audioContext.destination);
+      // PenguinMod thing
```

Comment:
i think you know how i feel about that; what does do for you?

Reply:
On penguinmod is allows the volume slider to affect playing songs. Turbowarp has no support :(

Reply:
> On penguinmod this allows the volume slider to affect playing songs. Turbowarp has no support :(

You can easily implement this by syncing the volume of the gain node in your audio context with the audio context of the project, that's not impossible.

Reply:
> > On penguinmod this allows the volume slider to affect playing songs. Turbowarp has no support :(
> 
> 
> 
> You can easily implement this by syncing the volume of the gain node in your audio context with the audio context of the project, that's not impossible.

How??